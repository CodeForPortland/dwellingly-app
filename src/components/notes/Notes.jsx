import React, { Component } from 'react';
import PropTypes from 'prop-types';
import MessageBox from '../MessageBox/MessageBox';

import './Notes.scss';

// TODO: Update "highlighted" ternary to check against currently logged-in user

class Notes extends Component {
  render() {
    const { action, notes, summary } = this.props;
    return (
      <div>
        {summary && (
          <div className="msgbox msgbox--inline">
            <h4>Notes</h4>
            <span className="count">{notes.length}</span>
          </div>
        )}
        {!summary && (
          <div className="notes">
            <header>
              <h4>Notes</h4>
              {action && action}
            </header>
            {notes &&
              notes.length > 0 &&
              notes.map(note => (
                <div key={note.id} className="note">
                  <div className="note__meta">
                    <p className="note__name">{note.name}</p>
                    <time className="node__date" dateTime={note.sent}>
                      {note.sent}
                    </time>
                  </div>
                  <MessageBox
                    user={note.name}
                    message={note.message} />
                </div>
              ))}
          </div>
        )}
      </div>
    );
  }
}

Notes.propTypes = {
  action: PropTypes.element,
  notes: PropTypes.arrayOf(PropTypes.shape({})),
  summary: PropTypes.bool
};

Notes.defaultProps = {
  action: undefined,
  notes: [],
  summary: false
};

export default Notes;
