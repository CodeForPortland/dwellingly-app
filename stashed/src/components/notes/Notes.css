/* V A R I A B L E S */
/* end variables */
/* F U N C T I O N S */
/* end functions */
/* M I X I N S */
/* end mixins */
/* N O T E S */
/* ==================================================== */
.notes {
  font-size: 0.875em;
}

.notes header {
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
      -ms-flex-pack: justify;
          justify-content: space-between;
}

.note {
  margin-top: 1em;
}

.note__meta {
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
      -ms-flex-pack: justify;
          justify-content: space-between;
  color: #858585;
}

.note__meta * {
  margin: 0;
}

/* end notes */
/*# sourceMappingURL=Notes.css.map */