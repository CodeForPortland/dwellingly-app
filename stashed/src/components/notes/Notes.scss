@import "../../constants/constants.scss";

/* N O T E S */
/* ==================================================== */
.notes {
  font-size: emify(14, 16);

  header {
    @include flex_align(space-between);
  }
}

.note {
  margin-top: 1em;

  &__meta {
    @include flex_align(space-between);
    color: $color__gray--medium;

    * {
      margin: 0;
    }
  }
}
/* end notes */
