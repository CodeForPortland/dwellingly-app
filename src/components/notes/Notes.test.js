import React from 'react';
import renderer from 'react-test-renderer';
import Notes from './Notes';

describe('Notes Component', () => {
  const props = {
    action: undefined,
    notes: [],
    summary: false,
  };

  it('renders div with className "notes" component', () => {
    const NotesComponent = renderer.create(<Notes {...props}/>).toJSON();
    expect(NotesComponent).toMatchSnapshot();
  });
  // it('renders div with className "msgbox msgbox--inline" if summary prop is true', () => {
  //   const props = {
  //     summary: true,
  //   };
  //   const NotesComponent = shallow(<Notes {...props}/>).find('span');
  //   expect(NotesComponent).toHaveLength(1);
  // });
  it('renders div with className="msgbox msgbox--inline" if summary prop is true', () => {
    const props = {
      summary: true,
    };
    const NotesComponent = renderer.create(<Notes {...props}/>).toJSON();
    expect(NotesComponent).toMatchSnapshot();
  });
  it('renders notes array if notes exist', () => {
    const props = {
      notes: [1],
    };
    const NotesComponent = renderer.create(<Notes {...props}/>).toJSON();
    expect(NotesComponent).toMatchSnapshot();
  });
});
