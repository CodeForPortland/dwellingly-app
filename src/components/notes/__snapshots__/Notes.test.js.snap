// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notes Component renders div with className="msgbox msgbox--inline" if summary prop is true 1`] = `
<div>
  <div
    className="msgbox msgbox--inline"
  >
    <h4>
      Notes
    </h4>
    <span
      className="count"
    >
      0
    </span>
  </div>
</div>
`;

exports[`Notes Component renders div with className="notes" if summary prop is false 1`] = `
<div>
  <div
    className="notes"
  >
    <header>
      <h4>
        Notes
      </h4>
    </header>
  </div>
</div>
`;
